quesma/table_resolver/rules.go

```diff
@@ -209,214 +251,168 @@ func (r *tableRegistryImpl) singleIndex(indexConfig map[string]config.IndexConfi
-					Reason:        "Only Elastic matched.",
+		if r.conf.AutodiscoveryEnabled {
+			for indexName, index := range r.clickhouseIndexes {
+				if index.isVirtual && indexName == part {
+					virtualTableExists = true
+					break
 				}
+			}
+		}
 
-			case nElastic == 0 && nClickhouse > 0:
-				// it will be resolved by sth else later
-				return nil
-
-			case nElastic == 0 && nClickhouse == 0:
-				return &Decision{
-					IsEmpty: true,
-					Reason:  "No indexes matched. Checked both connectors.",
-				}
+		if idxConfig, ok := cfg[part]; (ok && idxConfig.UseCommonTable) || (virtualTableExists) {
+			return &Decision{
+				UseConnectors: []ConnectorDecision{&ConnectorDecisionClickhouse{
+					ClickhouseTableName: common_table.TableName,
+					ClickhouseTables:    []string{part},
+					IsCommonTable:       true,
+				}},
+				Reason: "Common table will be used.",
 			}
 		}
 
 		return nil
 	}
-
 }
 
-func (r *tableRegistryImpl) makeCommonTableResolver(cfg map[string]config.IndexConfiguration, pipeline string) func(input parsedPattern) *Decision {
-
-	return func(input parsedPattern) *Decision {
-
-		if input.isPattern {
-
-			// At this point we should do not have any elastic indexes.
-			// This is because we have already checked if the pattern matches any elastic indexes.
-			for _, pattern := range input.parts {
-				for indexName := range r.elasticIndexes {
-					if util.IndexPatternMatches(pattern, indexName) {
-						return &Decision{
-							Err:    end_user_errors.ErrSearchCondition.New(fmt.Errorf("index parsedPattern [%s] resolved to elasticsearch indices", input.parts)),
-							Reason: "We're not supporting common tables for Elastic.",
-						}
-					}
+func mergeUseConnectors(lhs []ConnectorDecision, rhs []ConnectorDecision) ([]ConnectorDecision, *Decision) {
+	for _, connDecisionRhs := range rhs {
+		foundMatching := false
+		for _, connDecisionLhs := range lhs {
+			if _, ok := connDecisionRhs.(*ConnectorDecisionElastic); ok {
+				if _, ok := connDecisionLhs.(*ConnectorDecisionElastic); ok {
+					foundMatching = true
 				}
 			}
-
-			var matchedVirtualTables []string
-			var matchedTables []string
-			for _, pattern := range input.parts {
-
-				// here we check against the config
-
-				for indexName, index := range r.conf.IndexConfig {
-					if util.IndexPatternMatches(pattern, indexName) {
-
-						targets := getTargets(index, pipeline)
-
-						if slices.Contains(targets, config.ClickhouseTarget) {
-							if index.UseCommonTable {
-								matchedVirtualTables = append(matchedVirtualTables, indexName)
-							} else {
-								matchedTables = append(matchedTables, indexName)
-							}
+			if rhsClickhouse, ok := connDecisionRhs.(*ConnectorDecisionClickhouse); ok {
+				if lhsClickhouse, ok := connDecisionLhs.(*ConnectorDecisionClickhouse); ok {
+					if lhsClickhouse.ClickhouseTableName != rhsClickhouse.ClickhouseTableName {
+						return nil, &Decision{
+							Reason: "Inconsistent ClickHouse table usage",
```

Comment:
We should add more details here.

Reply:
Fixed in last commit - made the reason (and corresponding Err) more clear.